### ISSUE TITLE
Class-based generic views

### ISSUE THREAD

**Issue by andrew-snek** [kennethreitz/responder#235]
What do you think about adding something similar to Django's generic class-based Redirect- and TemplateViews?

These are examples of usage of what I've implemented so far:

```
import responder

api = responder.API()

@api.route('/redirect')
class ExampleRedirectView:
    redirect_url = '/hello/world'
    
    def on_request(self, req, resp):
        print("redirecting...")

@api.route('/{greeting}/{name}')
class ExampleTemplateView:
    template = 'hello.html' # file content: '<html>{{ salute }}</html>'

    def on_request(self, req, resp, *, greeting, name):
        print("generating salute...")
        resp.context = {
            'salute': f'{greeting}, my dear {name}'
        }

if __name__ == '__main__':
    api.run()
```

When you enter 'localhost:5042/redirect', you see "doing something..." in console, then "generating salute..." in console and finally 'hello, my dear world' on the screen. Also you don't need to inherit anything.

What do you, people, think? Is something like this needed?

**Comment by kobayashi** [comment]
This issue is older than ~14 days, and no discussion.

**Comment by CentryPlan** [comment]
Interesting, can a get and post request be done from one  class based view  and also can i define more than one end point ??

though the topic is closed this is the only place i've seen this approach mentioned.

**Comment by taoufik07** [comment]
@CentryPlan yes it can. each HTTP method will call the corresponding method in the CBV and all HTTP methods are supported e.g. GET -> on_get, POST -> on_post, PUT -> on_put...
